### PR TITLE
lib: memory leak, fixes #4925

### LINKED ISSUE #4925: lib: kdb.c memory leak
## Steps to Reproduce the Problem

```sh
kdb meta set spec:/tests/set mountpoint /tmp/spec.ni
kdb meta set spec:/tests/set/some/testKey1 type unsigned_short
kdb spec-mount /tests/set
kdb set -f user:/tests/set/some/testKey1 -5

# causes it
kdb validate user:/tests/set/some
```

Looks like the variables are not freed properly in case of an error:

https://github.com/ElektraInitiative/libelektra/blob/d8a5905ad3073934d0dc56f508bed0bb6012b996/src/libs/elektra/kdb.c#L1928-L1946C3?plain=1

## Expected Result
no memory leak

## Actual Result
memory leak

## System Information

- Elektra: master
- Operating System: Arch Linux

`valgrind-3.21.0`

### PR DESCRIPTION
**Pull request by hannes99**
#4925 

## Basics

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add them to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - add a line in `doc/news/_preparation_next_release.md`
  - reformat the code with `scripts/dev/reformat-all`
  - make all unit tests pass
  - fix all memleaks
  - fix the CI itself (or rebase if already fixed)
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if everything is done** and no further pushes are planned by you.

### CODE REVIEW

File: src/libs/elektra/kdb.c [hunk 1]
```diff
@@ -1934,14 +1934,18 @@ int kdbGet (KDB * handle, KeySet * ks, Key * parentKey)
 		ELEKTRA_SET_INTERNAL_ERROR (parentKey,
 					    "Couldn't divide keys into mountpoints before poststorage. Please report this bug at "
 					    "https://issues.libelektra.org.");
+		keyDel (defaultCutpoint);
```
Comment (atmaxinger) [inline src/libs/elektra/kdb.c:1937]: Wouldn't it be better if you delete `defaultCutpoint` and `defaults` after the label `error`?

**Review by atmaxinger — APPROVED** [review]
LGTM

**Comment by atmaxinger** [comment]
Please add something to the release notes tho.

File: src/libs/elektra/kdb.c (hunk 1 above)
Comment (hannes99) [reply src/libs/elektra/kdb.c:1937]: The problem is `goto error` can already happens before those two vars exist, so it's either doing it like this, or setting them to `NULL` at the beginning and checking for `NULL` after the label. I felt like this is the better solution. Am I missing something?

Comment (atmaxinger) [reply src/libs/elektra/kdb.c:1937]: No, it's alright.

**Comment by atmaxinger** [comment]
Thank you for this fix!

**Pull request merged**